File: server/Server/Models/Agency.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Server.Models;

public class Agency : EntityBase
{
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int NumberOfLicences { get; set; }

    public int AgencyStatusId { get; set; }
    public AgencyStatus? AgencyStatus { get; set; }
    public virtual IList<UserProfile> Users { get; set; } = new List<UserProfile>();
}

File: client/src/app/clients/models/agency.ts
export class Agency {
  id: number;
  name: string;
  description: string;
  numberOfLicences: number;
  agencyStatusId: number;
  rowVer: number;

  constructor(
    obj?: any
  ) {
    this.id = obj.id;
    this.name = obj.name;
    this.description = obj.description;
    this.numberOfLicences = obj.numberOfLicences;
    this.agencyStatusId = obj.agencyStatusId;
    this.rowVer = obj.rowVer;
  }
}
